add wildcard processing and check for EMPTY_RESULTS instead of null and add unit test

austin007008 opened the pull request:
add wildcard processing and check for EMPTY_RESULTS instead of null and add unit test

austin007008 committed:
add wildcard processing and check for EMPTY_RESULTS instead of null

drewfarris commented on warehouse/keyword-common/src/main/java/datawave/util/keyword/KeywordExtractor.java:
```diff
@@ -99,23 +102,37 @@ public void parseOptions(Map<String,String> iteratorOptions) {
         }
     }
 
+    @Nonnull
```
Consider adding the javadoc:
```suggestion
    /**
     * Extract keywords from the <code>foundContent</code> field in priority order based on the list of views in <code>preferredViews</code> field. Will return
     * a KeywordResults object populated with the keywords extracted from the first view in <code>preferredViews</code> that is found to contain content. If no
     * keywords can be extracted from all preferred views, returns an empty <code>KeywordResults</code> object.
     *
     * @return a KeywordResults object containing keywords from the first of the <code>preferredViews</code> found in <code>foundContent</code> that yields a
     *         non-empty set of keywords or an empty KeywordResults object if no keywords can be extracted.
     */
    @Nonnull
```

drewfarris commented on warehouse/keyword-common/src/main/java/datawave/util/keyword/KeywordExtractor.java:
```diff
@@ -99,23 +102,37 @@ public void parseOptions(Map<String,String> iteratorOptions) {
+        return EMPTY_RESULTS;
+    }
+
+    @Nonnull
```
Consider adding this javadoc:
```suggestion
    /**
     * Attempts to extract keywords from VisibleContent object provided.
     *
     * @param viewName
     *            the view name we're extracting from.
     * @param content
     *            the content associated with that view.
     * @return a KeywordResult object containing keywords extracted from the provided content, or the canonical <code>EMPTY_RESULTS</code> KeywordResults object
     *         if no keywords could be extracted.
     */
    @Nonnull
```

drewfarris reviewed (changes requested):
Just some minor javadoc additions for clarity.

Thanks for catching the use of the guava apis we shouldn't be using any longer.

austin007008 committed:
Apply suggestions from code review

Add additional javadocs for extractKeywords and extractkeywordsFromVisableContent methods

Co-authored-by: Drew Farris <drew.farris@gmail.com>

austin007008 committed:
add javadocs for KywordExtractingiteratortest

austin007008 committed:
format
